spec/operators/switchScan-spec.ts

```diff
@@ -0,0 +1,398 @@
+import { expect } from 'chai';
+import { concat, defer, Observable, of } from 'rxjs';
+import { hot, cold, expectObservable, expectSubscriptions } from '../helpers/marble-testing';
+import { switchScan, map, mergeMap, takeWhile } from 'rxjs/operators';
+
+/** @test {switchScan} */
+describe('switchScan', () => {
+  it('should map-and-flatten each item to an Observable while passing the accumulated value', () => {
+    const e1 =    hot('--1-----3--5-------|');
+    const e1subs =    '^                  !';
+    const e2 =   cold('x-x-x|              ', {x: 10});
+    const expected =  '--x-x-x-y-yz-z-z---|';
+    const values = {x: 10, y: 40, z: 90};
+
+    const result = e1.pipe(switchScan((acc, x) => e2.pipe(map(i => i * Number(x) + acc)), 0));
+
+    expectObservable(result).toBe(expected, values);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should not pass seed when undefined', () => {
```

Comment:
nitpick: This test description doesn't make much sense. The seed passed is zero and the expectation tests the accumulator  not the seed.

Reply:
Yeah, now that I look at it, the test is overly complicated as well. I've simplified it a bit.